File: sidecars/synaplan-compute/tests/hostile/node/disk_fill.js
// expected-result: failed
// reason: output_limit
// note: Stops at the fsize ulimit / outputMb cap; host disk unchanged
// disk_fill.js — write until ENOSPC, EFBIG, or the output cap.
const fs = require('fs');
const chunk = Buffer.alloc(1024 * 1024, 65);
let written = 0;
try {
  const fd = fs.openSync('/work/fill.bin', 'w');
  while (written <= 512 * 1024 * 1024) {
    fs.writeSync(fd, chunk);
    written += chunk.length;
  }
  fs.closeSync(fd);
} catch (e) {
  process.stderr.write('stopped: ' + e.message + '\n');
  process.exit(1);
}
process.stderr.write('filled without hitting a cap\n');
process.exit(1);

File: sidecars/synaplan-compute/tests/hostile/disk_fill.py
# expected-result: failed
# reason: output_limit
# note: Stops at the /work /tmp size caps; host disk unchanged
"""disk_fill.py — write until ENOSPC or the output ulimit."""
import os
import sys

path = "/work/fill.bin"
chunk = b"A" * (1024 * 1024)
written = 0
try:
    with open(path, "wb") as f:
        while True:
            f.write(chunk)
            written += len(chunk)
            f.flush()
            if written > 512 * 1024 * 1024:
                break
except OSError as exc:
    print("stopped:", exc, file=sys.stderr)
    sys.exit(1)
print("filled without hitting a cap", file=sys.stderr)
sys.exit(1)
